balrog/os-de-support.rst

```diff
@@ -0,0 +1,198 @@
+
+    "Darwin_aarch64-gcc3": {
+      "alias": "Darwin_x86_64-gcc3-u-i386-x86_64"
+    },
+    "Darwin_x86-gcc3": {
+      "alias": "Darwin_x86_64-gcc3-u-i386-x86_64"
+    },
+    "Darwin_x86-gcc3-u-i386-x86_64": {
+      "alias": "Darwin_x86_64-gcc3-u-i386-x86_64"
+    },
+    "Darwin_x86_64-gcc3": {
+      "alias": "Darwin_x86_64-gcc3-u-i386-x86_64"
+    },
+    "WINNT_x86-msvc-x64": {
+      "alias": "WINNT_x86-msvc"
+    },
+    "WINNT_x86-msvc-x86": {
+      "alias": "WINNT_x86-msvc"
+    },    
+    "WINNT_x86_64-msvc-x64": {
+      "alias": "WINNT_x86_64-msvc"
+    },    
+
+* Create a new De-Support release referencing the SUMO KB article. Careful: Use a locale-agnostic link like https://support.mozilla.org/kb/firefox-users-macos-1012-1013-1014-moving-to-extended-support (instead of https://support.mozilla.org/en-US/kb/firefox-users-macos-1012-1013-1014-moving-to-extended-support). Example release blob: ::
+
+    {
+        "detailsUrl": "https://support.mozilla.org/kb/firefox-users-macos-1012-1013-1014-moving-to-extended-support",
+        "displayVersion": "115.0",
+        "name": "OSX-10.12-10.14-Desupport",
+        "product": "Firefox",
+        "schema_version": 50
+    }
+
+* Create a rule to serve the de-support notice: match the ``Firefox : Nightly*`` channel and the affected OS version with lower priority than the pinning rule (so probably those who have been updated to the last supported build); serve the new De-Support release.
+* Delete the ``NoUpdate`` rule created earlier.
+
+Update Balrog rules for Beta and Aurora
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Check with the *Product* organization to verify per-channel requirements for Beta and DevEdition.
+
+To implement DE-SUPPORT for DevEdition, follow the de-support procedure for Nightly, above, with these changes:
+
+* Make changes just before and after the Merge Day II merge of central to beta.
+* Use the ``Firefox : aurora*`` channel
+* There should be no need to add aliases to the release blob.
+
+To implement ESR-SWITCH for Beta, follow the de-support procedure for Release, below, with these changes:
+
+* Make changes just before and after the Merge Day II merge of central to beta.
+* Use the ``Firefox : beta*`` channel
+* There should be no need to add aliases to the release blob.
+
+Update Balrog rules for Release
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For the Release channel, we generally ESR-SWITCH: serve an update referencing the channel-switching MAR, so that
+users on the affected OS move from the Release channel to the ESR channel.
+
+Just *before* the first release build of the first unsupported version, pause Release updates for the affected OS version. This will typically be just before the Merge Day I merge of Beta to Release, for the release after the Beta changes.
```

Comment:
Probably want to change this section so that:
- ahead of time, create the esr-switch release blob, with a pointer to your script
- at the start of RC week, set up the ESR-SWITCH rule on `beta-localtest` and `release-localtest`
- we get QA to verify the migration
- on merge day II, set it up on `beta-cdntest` and `release-cdntest`
- on release day, set it up for `beta*` and `release*`
?